optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/drools/DroolsConstraint.java

```diff
@@ -164,6 +92,19 @@ public String getConstraintName() {
         return constraintName;
     }
 
+    public int getConstraintStreamCardinality() {
+        if (scoringStream instanceof UniConstraintStream) {
+            return 1;
+        } else if (scoringStream instanceof BiConstraintStream) {
+            return 2;
+        } else if (scoringStream instanceof TriConstraintStream) {
+            return 3;
```

Comment:
Any idea what PolyConstraintStream will return? Int.MAX?

Reply:
This is a solution to the justification list problem - I think we'll have to solve that before poly streams.